src/Widgets/WindowClone.vala

```diff
@@ -409,10 +409,16 @@ namespace Gala {
                 window_icon.opacity = 0;
                 set_window_icon_position (outer_rect.width, outer_rect.height);
 
-                window_icon.get_transition ("opacity").completed.connect (() => {
+                var transition = window_icon.get_transition ("opacity");
+                if (transition != null) {
+                    transition.completed.connect (() => {
+                        in_slot_animation = false;
+                        place_widgets (outer_rect.width, outer_rect.height);
+                    });
+                } else {
                     in_slot_animation = false;
                     place_widgets (outer_rect.width, outer_rect.height);
-                });
+                }
```

Comment:
NOTE: I allowed myself to push another tiny fix that now finally let's gala start assertion-failure free on my end. On WM startup, the icons already have 0-opacity, as such there is no transition to be had.

Reply:
This patch looks good to me, thanks!